packages/flutter_tools/lib/src/asset.dart

```diff
@@ -401,6 +405,8 @@ Map<_Asset, List<_Asset>> _parseAssets(
   return result;
 }
 
+final RegExp _leadingParentRegExp = new RegExp(r'(\.\.\/)+');
```

Comment:
This doesn't handle the case of "foo/../../../../bar/baz"

Does https://www.dartdocs.org/documentation/path/1.4.1/path/normalize.html not do what we want?  Maybe we need to resolve it against a base path of `/` or something in addition.